src/Websocket.Client/WebsocketClient.cs

```diff
@@ -23,7 +23,7 @@ public class WebsocketClient : IWebsocketClient
         private readonly WebsocketAsyncLock _locker = new WebsocketAsyncLock();
         private Uri _url;
         private Timer _lastChanceTimer;
-        private readonly Func<WebSocket> _clientFactory;
+        private readonly Func<Uri, CancellationToken, Task<WebSocket>> _connectionFactory;
```

Comment:
Nothing with this line, but please revert back public interface property `NativeClient` to: 
`public ClientWebSocket NativeClient => _client as ClientWebSocket;`

Reply:
This is now fixed.

But I think you should give an error when trying to get the NativeClient?

Reply:
Yes, that would be more explicit

Reply:
I have now changed it to be a normal cast and therefore throw an InvalidCastException.

Reply:
Thank you, merging PR.. 
I will change that to some custom exception with the message, I hate that default .NET exceptions, very confusing in runtime.